src/Workspaces/CoreTest/SolutionTests/SolutionTests.cs

```diff
@@ -1554,6 +1554,57 @@ public void TestWithSyntaxTree()
             var root = newTree.GetRoot();
         }
 
+        [Fact, Trait(Traits.Feature, Traits.Features.Workspace)]
+        [WorkItem(3705, "https://github.com/dotnet/roslyn/issues/3705")]
+        public void TestAnalyzerConfigOptions()
+        {
+            // get one to get to syntax tree factory
+            var dummyProject = CreateNotKeptAliveSolution().AddProject("dummy", "dummy", LanguageNames.CSharp);
+
+            var factory = dummyProject.LanguageServices.SyntaxTreeFactory;
+
+            // create the origin tree
+            var testDir = Temp.CreateDirectory();
+            var sourceFile = testDir.CreateFile("test.cs").WriteAllText(@"
+class C
+{
+    void M(C? c)
+    {
+        _ = c.ToString();   // warning CS8602: Dereference of a possibly null reference.
+    }
+}");
+            using var fileStream = File.OpenRead(sourceFile.Path);
+            var text = SourceText.From(fileStream);
+            var tree = factory.ParseSyntaxTree(sourceFile.Path, dummyProject.ParseOptions, text, analyzerConfigOptionsResult: null, CancellationToken.None);
+
+            var compilation = CSharpCompilation.Create("TestProject", new[] { tree }, references: new[] { TestReferences.NetFx.v4_0_30319.mscorlib },
+                    options: new CSharpCompilationOptions(OutputKind.DynamicallyLinkedLibrary).WithNullableContextOptions(NullableContextOptions.Enable));
+
+            // warning CS8602: Dereference of a possibly null reference.
+            var diagnostics = compilation.GetDiagnostics();
+            Assert.Equal(1, diagnostics.Length);
```

Comment:
```suggestion
            Assert.Single(diagnostics);
```

Reply:
(note this also returns the single object too, which is handy for the later assertions if you want)